superset/assets/src/visualizations/table.js

```diff
@@ -1,58 +1,108 @@
 import d3 from 'd3';
+import PropTypes from 'prop-types';
+import $ from 'jquery';
 import dt from 'datatables.net-bs';
 import 'datatables.net-bs/css/dataTables.bootstrap.css';
 import dompurify from 'dompurify';
-
 import { fixDataTableBodyHeight, d3TimeFormatPreset } from '../modules/utils';
 import './table.css';
 
-const $ = require('jquery');
-
 dt(window, $);
 
-function tableVis(slice, payload) {
-  const container = $(slice.selector);
-  const fC = d3.format('0,000');
+const propTypes = {
+  data: PropTypes.shape({
+    records: PropTypes.arrayOf(PropTypes.object),
+    columns: PropTypes.arrayOf(PropTypes.string),
+  }),
+  height: PropTypes.number,
+  alignPn: PropTypes.bool,
+  colorPn: PropTypes.bool,
+  columnFormats: PropTypes.object,
+  filters: PropTypes.object,
+  includeSearch: PropTypes.bool,
+  metrics: PropTypes.arrayOf(PropTypes.string),
+  onAddFilter: PropTypes.func,
+  onRemoveFilter: PropTypes.func,
+  orderDesc: PropTypes.bool,
+  pageLength: PropTypes.oneOfType([
+    PropTypes.number,
+    PropTypes.string,
+  ]),
+  percentMetrics: PropTypes.array,
+  tableFilter: PropTypes.bool,
+  tableTimestampFormat: PropTypes.string,
+  timeseriesLimitMetric: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.object,
+  ]),
+  verboseMap: PropTypes.object,
+};
+
+const fC = d3.format('0,000');
+function NOOP() {}
+
+function TableVis(element, props) {
+  PropTypes.checkPropTypes(propTypes, props, 'prop', 'TableVis');
+
+  const {
+    data,
+    height,
+    alignPn,
+    colorPn,
+    columnFormats,
+    filters,
+    includeSearch,
+    metrics: rawMetrics,
+    onAddFilter = NOOP,
```

Comment:
Why do we need a No Op on these prop functions?

Reply:
To avoid checking if `onAddFilter` exists every time before calling it.